addons/auth_passkey/models/key.py

```diff
@@ -0,0 +1,146 @@
+import base64
+from werkzeug.urls import url_parse
+
+from odoo import api, fields, models, _
+from odoo.tools import sql
+from odoo.addons.base.models.res_users import check_identity
+
+from ..lib.duo_labs.webauthn import base64url_to_bytes, generate_authentication_options, verify_authentication_response, generate_registration_options, verify_registration_response
+from ..lib.duo_labs.webauthn.helpers.structs import AuthenticatorSelectionCriteria, ResidentKeyRequirement, UserVerificationRequirement
+
+
+class PassKey(models.Model):
+    _name = 'auth.passkey.key'
+    _description = 'Passkey'
+    _order = 'id desc'
+
+    name = fields.Char(required=True)
+    credential_identifier = fields.Char(required=True, groups='base.group_system')
+    public_key = fields.Char(required=True, groups='base.group_system', compute='_compute_public_key', inverse='_inverse_public_key')
+    sign_count = fields.Integer(default=0, groups='base.group_system')
+
+    _sql_constraints = [
+        ('unique_identifier', 'UNIQUE(credential_identifier)', 'The credential identifier should be unique.'),
+    ]
+
+    def init(self):
+        super().init()
+        if not sql.column_exists(self.env.cr, self._table, 'public_key'):
+            self.env.cr.execute('ALTER TABLE %s ADD COLUMN public_key varchar' % self._table)
+
+    def _compute_public_key(self):
+        query = 'SELECT public_key FROM %s WHERE id = %%s' % self._table
+        for passkey in self:
+            self.env.cr.execute(query, [passkey.id])
```

Comment:
Use the new SQL wrapper `odoo.tools.SQL`

```suggestion
            self.env.cr.execute(SQL('SELECT public_key FROM %s WHERE id = %s', SQL.identifier(self._table), passkey.id))
```

This is 100% safe against SQL injection, even if the attacker finds a way to control `self._table`.
While the former version is not safe against this.